syntax = "proto3";
package chat;
import "google/protobuf/any.proto";

message NetMsg {
    string token = 1;
    google.protobuf.Any data = 2;
}

message Response {
    bool success = 1;
    string info = 2;
}

message LoginRequest {
    string username = 1;
    string password = 2;
}
message LoginResponse {
    bool success = 3;
    string info = 4;
    int32 id = 1;
    string token = 2;
}

message SignupRequest {
    string username = 1;
    string password = 2;
}

message LogoutRequest {
    int32 senderID = 1;
}

message GetUserInfosRequest {
    int32 senderID = 1;
}
message GetUserInfosResponse {
    repeated UserInfo users = 1;
}

message GetMessagesRequest {
    int32 senderID = 1;
    int32 afterTimeUnix = 2;
}
message GetMessagesResponse {
    repeated ChatMessage messages = 1;
}

message MakeFriendRequest {
    int32 senderID = 1;
    int32 targetID = 2;
}

message ChatMessage {
    int32 senderID = 1;
    int32 targetID = 2;
    int32 timeUnix = 3;

    oneof Content {
        string text = 10;
        bytes file = 11;
    }
}

message UserInfo {
    int32 id = 1;
    string username = 2;

    bool isFriend = 10;
    bool isOnline = 11;
}